drf_jsonapi/serializers/resources.py

```diff
@@ -435,7 +435,8 @@ def sort(cls, sort_param=None, queryset=None):
         sort_fields = list(filter(None, sort_param.split(',')))
 
         # validate the sort fields actually exist in the model
-        field_names = [field.name for field in queryset.model._meta.get_fields()]
+        field_names = getattr(cls.Meta, 'sort_fields', cls.Meta.fields)
+        field_names += ('id',)
```

Comment:
Just curious - why the need to add `id` back in here?

Reply:
2 reasons. 1) We have tests that assume you can sort by ID so they fail without 'id' and 2) We now check Serializer.Meta.fields for sortable fields by default but that NEVER contains id because id is always included on the serialized output.